lib/Runtime/Types/DynamicObjectPropertyEnumerator.cpp

```diff
@@ -255,13 +255,18 @@ namespace Js
             }
         } while (Js::IsInternalPropertyId(propertyId));
 
-        if (info.GetPropertyString() != nullptr && info.GetPropertyString()->ShouldUseCache() && propertyString == info.GetPropertyString())
+        if (info.GetPropertyRecordUsageCache() != nullptr && info.GetPropertyRecordUsageCache()->ShouldUseCache())
         {
-            CacheOperators::CachePropertyRead(startingObject, this->object, false, propertyId, false, &info, scriptContext);
-            if ((!(this->flags & EnumeratorFlags::EphemeralReference)) && info.IsStoreFieldCacheEnabled() && info.IsWritable() && ((info.GetFlags() & (InlineCacheGetterFlag | InlineCacheSetterFlag)) == 0))
+            PropertyString* propertyStringAsPropertyString = propertyString == nullptr ? nullptr : PropertyString::TryFromVar(propertyString);
```

Comment:
When would this be null if we have a cache and we should be using it?

Reply:
It looked to me like propertyString could be null if the flags said to fetch Symbol properties, though we may never use this method with those flags.

---
In reply to: [162228454](https://github.com/Microsoft/ChakraCore/pull/4569#discussion_r162228454) [](ancestors = 162228454)